repository_service_tuf_api/__init__.py

```diff
@@ -185,16 +163,44 @@ def sync_redis():
 # https://github.com/repository-service-tuf/repository-service-tuf-api/issues/91
 
 
+def pre_lock_bootstrap(task_id):
+    """
+    Add a pre-lock to the bootstrap repository settings.
+
+    Add to the repository settings in Redis the lock as `pre-<task_id>`.
+
+    Args:
+        task_id: Task id generated by bootstrap
+    """
+    settings_data = settings_repository.as_dict(
+        env=settings_repository.current_env
+    )
+    settings_data["BOOTSTRAP"] = f"pre-{task_id}"
+    redis_loader.write(settings_repository, settings_data)
+
+
+def release_bootstrap_lock():
+    """
+    Remove the pre-lock from repository settings.
+
+    Move the repository settings BOOTSTRAP to None if not finished.
+    """
+    settings_data = settings_repository.as_dict(
+        env=settings_repository.current_env
+    )
+    settings_data["BOOTSTRAP"] = None
+    redis_loader.write(settings_repository, settings_data)
+
+
 def is_bootstrap_done():
     """
     Check if the boot is done.
     """
-
-    sync_redis()
-    if settings_repository.get_fresh("BOOTSTRAP", False):
-        return True
-    else:
+    settings_repository.reload()  # reload the settings
+    if settings_repository.get_fresh("BOOTSTRAP", None) is None:
```

Comment:
Why `reload` is required if we call `settings_repository.get_fresh()`?

Reply:
Added better comment